app/Http/Controllers/Api/V1/IndependentActivityController.php

```diff
@@ -969,4 +970,52 @@ public function copyIndependentActivityIntoPlaylist(Request $request, Organizati
             received and is being processed.<br> You will be alerted in the notification section in the title bar when complete.",
         ], 200);
     }
+
+    /**
+     * Move Independent Activity into Playlist
+     *
+     * Move the specified independent activity of an suborganization and link with a playlist.
+     *
+     * @urlParam suborganization required The Id of a suborganization Example: 1
+     * @bodyParam ind_act query string comma seperated independent activities ids  Example: 1,2,3
+     * @urlParam playlist required The Id of a playlist Example: 1
+     *
+     * @response {
+     *   "message": "Independent Activity is being copied in background!"
+     * }
+     *
+     * @response 400 {
+     *   "errors": [
+     *     "Not a Public Independent Activity."
+     *   ]
+     * }
+     *
+     * @response 500 {
+     *   "errors": [
+     *     "Failed to copy independent activity."
+     *   ]
+     * }
+     *
+     * @param Request $request
+     * @param Organization $suborganization
+     * @param IndependentActivity $independent_activity
+     * @param Playlist $playlist
+     * @return Response
+     */
+    public function moveIndependentActivityIntoPlaylist(Request $request, Organization $suborganization, Playlist $playlist)
+    {
+        $ind_act = $request->get('ind_act');
+        $arr = explode(',',$ind_act); 
+        for ($i=0; $i < count($arr); $i++) {
```

Comment:
Shouldn't we move the loop to job, 
this will ensure that we roll back if any activity copy failure.